modules/apps/translation/translation-web/src/main/resources/META-INF/resources/js/translate/Translate.js

```diff
@@ -0,0 +1,320 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLayout from '@clayui/layout';
+import ClayLink from '@clayui/link';
+import {ClassicEditor} from 'frontend-editor-ckeditor-web';
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import TranslateLanguagesSelector from './TranslateLanguagesSelector';
+
+const noop = () => {};
+
+const TranslationFieldEditor = ({
+	editorConfiguration,
+	id,
+	label,
+	sourceContent,
+	sourceContentDir,
+	targetContent,
+	targetContentDir,
+	onChange = noop,
+}) => {
+	const [value, setValue] = useState(targetContent);
+
+	return (
+		<ClayLayout.Row>
+			<ClayLayout.Col md={6}>
+				<ClayForm.Group>
+					<label className="control-label">{label}</label>
+					<div
+						className="translation-editor-preview"
+						dangerouslySetInnerHTML={{__html: sourceContent}}
+						dir={sourceContentDir}
+					/>
+				</ClayForm.Group>
+			</ClayLayout.Col>
+			<ClayLayout.Col md={6}>
+				<ClayForm.Group>
+					<label className="control-label">{label}</label>
+					<ClassicEditor
+						contents={targetContent}
+						editorConfig={{
+							...editorConfiguration.editorConfig,
+							contentsLangDirection: targetContentDir,
+						}}
+						name={id}
+						onChange={(data) => {
+							if (value !== data.trim()) {
+								setValue(data);
+								onChange(data);
+							}
+						}}
+					/>
+					<input defaultValue={value} name={id} type="hidden" />
+				</ClayForm.Group>
+			</ClayLayout.Col>
+		</ClayLayout.Row>
+	);
+};
+
+const TranslationFieldInput = ({
+	id,
+	label,
+	multiline,
+	sourceContent,
+	sourceContentDir,
+	targetContent,
+	targetContentDir,
+	onChange = noop,
+}) => {
+	const [value, setValue] = useState(targetContent);
+
+	return (
+		<ClayLayout.Row>
+			<ClayLayout.Col md={6}>
+				<ClayForm.Group>
+					<label className="control-label">{label}</label>
+					<ClayInput
+						component={multiline ? 'textarea' : undefined}
+						defaultValue={sourceContent}
+						dir={sourceContentDir}
+						readOnly
+						type="text"
+					/>
+				</ClayForm.Group>
+			</ClayLayout.Col>
+			<ClayLayout.Col md={6}>
+				<ClayForm.Group>
+					<label className="control-label" htmlFor={id}>
+						{label}
+					</label>
+					<ClayInput
+						component={multiline ? 'textarea' : undefined}
+						dir={targetContentDir}
+						id={id}
+						name={id}
+						onChange={(event) => {
```

Comment:
I prefer to extract these stuff to a method, but I guess is a matter of taste.

Reply:
I´m moving nav into a component and move this into a function. 👍